config/routes.rb

```diff
@@ -3,4 +3,10 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  root "users#index"
+
+  get "/users", to: "users#index", as: "users"
+  get "/users/:id", to: "users#show", as: "user"
+  get "/users/:user_id/posts", to:"posts#index", as: "user_posts"
+  get "/users/:user_id/posts/:id", to: "posts#show", as: "user_post"
```

Comment:
- [ ] [OPTIONAL] I think it would be better to use [resource routing](https://guides.rubyonrails.org/v3.2.2/routing.html#resource-routing-the-rails-default) for defining routes. It not only simplifies your code but also aligns with Rails conventions, making your application more maintainable and developer-friendly.


```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
ok, i change and use resource routing, thank you